### PR TITLE
pkg: update littlefs pkg to v1.3

### PR DESCRIPTION
**Pull request by vincent-d**
### Contribution description

Update package to a tagged version (v1.3).
Also add license field and clean up Makefile

### Issues/PRs references

None

### CODE REVIEW

File: pkg/littlefs/Makefile [hunk 1]
```diff
@@ -1,15 +1,12 @@
 PKG_NAME=littlefs
 PKG_URL=https://github.com/geky/littlefs.git
-PKG_VERSION=be22d3449f23a24e7462114349ce04d751e42437
+PKG_VERSION=v1.3
 PKG_BUILDDIR ?= $(PKGDIRBASE)/$(PKG_NAME)
```
Comment (aabadie) [inline pkg/littlefs/Makefile]: Maybe the PKG_BUILDDIR line could also be removed. I think it's set automatically.

**Comment by vincent-d** [comment]
Sorry, I realized that new error codes have been added, so I pushed a new commit.

File: pkg/littlefs/Makefile (hunk 1 above)
Comment (vincent-d) [reply pkg/littlefs/Makefile]: Amended directly

**Comment by vincent-d** [comment]
> works on native. But fails on boards samr21-xpro and pba-d-01-kw2x

It seems to come from the 'dummy' mtd implemented in the unittests. I'll take a look, but anyway this is not related to this PR.

**Comment by vincent-d** [comment]
@smlng See #8607

**Comment by vincent-d** [comment]
@gebart does your ACK hold?

File: pkg/littlefs/fs/littlefs_fs.c [hunk 2]
```diff
@@ -48,6 +48,10 @@ static int littlefs_err_to_errno(ssize_t err)
         return -ENOTDIR;
     case LFS_ERR_ISDIR:
         return -EISDIR;
+    case LFS_ERR_NOTEMPTY:
+        return -ENOTEMPTY;
+    case LFS_ERR_BADF:
+        return - EBADF;
```
Comment (gebart) [inline pkg/littlefs/fs/littlefs_fs.c]: remove the extra space

**Comment by vincent-d** [comment]
> Looks fine, except for the extra space in -EBADF

amended directly

**Pull request merged**